cloud/deployment/deployment.go

```diff
@@ -605,85 +716,28 @@ func HealthPoll(deploymentID, ws string, sleepTime, tickNum, timeoutNum int, cor
 	}
 }
 
-func Update(deploymentID, label, ws, description, deploymentName, dagDeploy, executor, schedulerSize, highAvailability string, schedulerAU, schedulerReplicas int, wQueueList []astro.WorkerQueue, forceDeploy bool, enforceCD *bool, client astro.Client) error { //nolint
+func Update(deploymentID, name, ws, description, deploymentName, dagDeploy, executor, schedulerSize, highAvailability, cicdEnforcement, defaultTaskPodCpu, defaultTaskPodMemory, resourceQuotaCpu, resourceQuotaMemory string, schedulerAU, schedulerReplicas int, wQueueList []astroplatformcore.WorkerQueueRequest, hybridQueueList []astroplatformcore.HybridWorkerQueueRequest, newEnvironmentVariables []astroplatformcore.DeploymentEnvironmentVariableRequest, force bool, coreClient astrocore.CoreClient, platformCoreClient astroplatformcore.CoreClient) error { //nolint
 	var queueCreateUpdate, confirmWithUser bool
 	// get deployment
-	currentDeployment, err := GetDeployment(ws, deploymentID, deploymentName, false, client, nil)
+	currentDeployment, err := GetDeployment(ws, deploymentID, deploymentName, false, platformCoreClient, nil)
 	if err != nil {
 		return err
 	}
-
-	configOption, err := client.GetDeploymentConfig()
+	c, err := config.GetCurrentContext()
 	if err != nil {
 		return err
 	}
-
-	// build query input
-	scheduler := astro.Scheduler{}
-
-	if schedulerAU != 0 {
-		scheduler.AU = schedulerAU
-	} else {
-		schedulerAU = currentDeployment.DeploymentSpec.Scheduler.AU
-		scheduler.AU = currentDeployment.DeploymentSpec.Scheduler.AU
-	}
-
-	if schedulerReplicas != 0 {
-		scheduler.Replicas = schedulerReplicas
-	} else {
-		schedulerReplicas = currentDeployment.DeploymentSpec.Scheduler.Replicas
-		scheduler.Replicas = currentDeployment.DeploymentSpec.Scheduler.Replicas
-	}
-
-	spec := astro.DeploymentCreateSpec{
-		Scheduler: scheduler,
-		Executor:  currentDeployment.DeploymentSpec.Executor,
-	}
-
-	// change the executor if requested
-	confirmWithUser, spec = mutateExecutor(executor, spec, len(currentDeployment.WorkerQueues))
-
-	deploymentUpdate := &astro.UpdateDeploymentInput{
-		ID:             currentDeployment.ID,
-		ClusterID:      currentDeployment.Cluster.ID,
-		DeploymentSpec: spec,
-	}
-	if label != "" {
-		deploymentUpdate.Label = label
-	} else {
-		deploymentUpdate.Label = currentDeployment.Label
+	var isCicdEnforced bool
+	if cicdEnforcement == "" {
+		isCicdEnforced = currentDeployment.IsCicdEnforced
 	}
-	if description != "" {
-		deploymentUpdate.Description = description
-	} else {
-		deploymentUpdate.Description = currentDeployment.Description
+	if cicdEnforcement == disable {
+		isCicdEnforced = false
 	}
-
-	if enforceCD == nil {
-		deploymentUpdate.APIKeyOnlyDeployments = currentDeployment.APIKeyOnlyDeployments
-	} else {
-		deploymentUpdate.APIKeyOnlyDeployments = *enforceCD
+	if cicdEnforcement == enable {
+		isCicdEnforced = true
 	}
```

Comment:
nit, can shorten this a bit but still keep the readability

```suggestion
    var isCicdEnforced bool
    if cicdEnforcement == "" {
        isCicdEnforced = currentDeployment.IsCicdEnforced
    } else {
        isCicdEnforced = cicdEnforcement == enable
    }
```

Reply:
isCicdEnforced is a bool and cicdEnforcement is a string

Reply:
This is because update needs the ability for the user to indicate they want no change